lucene/core/src/java/org/apache/lucene/codecs/lucene99/Lucene99HnswVectorsReader.java

```diff
@@ -209,17 +210,17 @@ public static VectorEncoding readVectorEncoding(DataInput input) throws IOExcept
 
   private FieldEntry readField(IndexInput input, FieldInfo info) throws IOException {
     VectorEncoding vectorEncoding = readVectorEncoding(input);
-    VectorSimilarityFunction similarityFunction = readSimilarityFunction(input);
-    if (similarityFunction != info.getVectorSimilarityFunction()) {
+    VectorSimilarityFunction vectorSimilarityFunction = readSimilarityFunction(input);
+    if (vectorSimilarityFunction != info.getVectorSimilarityFunction()) {
       throw new IllegalStateException(
           "Inconsistent vector similarity function for field=\""
               + info.name
               + "\"; "
-              + similarityFunction
+              + vectorSimilarityFunction
               + " != "
               + info.getVectorSimilarityFunction());
     }
-    return FieldEntry.create(input, vectorEncoding, info.getVectorSimilarityFunction());
+    return FieldEntry.create(info, input, vectorEncoding, info.getVectorSimilarityFunction());
```

Comment:
Having both `info` and `info.getVectorSimilarityFunction()` jumps out here.
```suggestion
    return FieldEntry.create(info, input, vectorEncoding);
```

or

```suggestion
    return FieldEntry.create(info, input, vectorEncoding, vectorSimilarityFunction);
```